FCCee/IDEA/compact/IDEA_o2_v01/DREndcapTubes_o1_v01.xml

```diff
@@ -0,0 +1,278 @@
+      <!--<property name="RINDEX" ref="RI_DR_Polystyrene"/>-->
+      <!--<property name="ABSLENGTH" ref="AbsLen_DR_Polystyrene"/>-->
+    </material>
+    <!--clad S material-->
+    <material name="PMMA_Scin" >
+      <D type="density" unit="g/cm3" value="1.1901" />
+      <composite n="5" ref="C" />
+      <composite n="2" ref="O" />
+      <composite n="8" ref="H" />
+      <!--<property name="RINDEX" ref="RI_PMMA"/>-->
+      <!--<property name="ABSLENGTH" ref="AbsLen_PMMA"/>-->
+    </material>
+    <!--core C material-->
+    <material name="PMMA" >
+      <D type="density" unit="g/cm3" value="1.19" />
+      <composite n="5" ref="C" />
+      <composite n="2" ref="O" />
+      <composite n="8" ref="H" />
+      <property name="RINDEX" ref="RI_PMMA"/>
+      <property name="ABSLENGTH" ref="AbsLen_PMMA"/>
+    </material>
+    <!--clad C material-->
+    <material name="Fluorinated_Polymer">
+      <D type="density" unit="g/cm3" value="1.43" />
+      <composite n="2" ref="C" />
+      <composite n="2" ref="F" />
+      <property name="RINDEX" ref="RI_Fluorinated_Polymer"/>
+    </material>
+  </materials>
+
+  <!-- Define the detector. A detector has a name, type to be returned in .cpp file with 
+       DECLARE_DETELEMENT(type,create_detector_c++method), vis attributes (optional),
+       and sensitive-detector (true/false), if sensitive="true" a readout must the passed.
+       If a detector contains sub-detectors, their information must be included as below.
+  -->
+  <detectors>
+    <detector name="DREndcapTubes" type="DREndcapTubes" vis="DRETAssemblyVis" id ="DetID_DREndcapTubes" readout="DREndcapTubesRO">
+    <type_flags type=" DetType_CALORIMETER + DetType_ELECTROMAGNETIC + DetType_BARREL"/>
+    <dimensions 
+      inner_radius="DRETinnerRadius"
+      z_length="DRETtowerHeight"
+      deltaphi="DRETNbOfZRot"/>
+      <assembly material="Air" vis="DRETAssemblyVis"/>
+      <stave material="Air" vis="DRETStaveVis"/>
+      <tower material="Air" vis="DRETTowerVis"/>
+      <tube_S material="Brass" outer_radius="DRETTubeRadius" sensitive="false" vis="DRETtube_S_Vis"/>
+      <tube_C material="Brass" outer_radius="DRETTubeRadius" sensitive="false" vis="DRETtube_C_Vis"/>
+      <clad_S material="PMMA_Scin" outer_radius="DRETCladRadius" sensitive="false" vis="DRETclad_S_Vis"/>
+      <clad_C material="Fluorinated_Polymer" outer_radius="DRETCladRadius" sensitive="false" vis="DRETclad_C_Vis"/>
+      <core_S material="DR_Polystyrene" outer_radius="DRETCoreRadius" sensitive="false" vis="DRETcore_S_Vis"/>
+      <core_C material="PMMA" outer_radius="DRETCoreRadius" sensitive="false" vis="DRETcore_C_Vis"/>
+    </detector>
+  </detectors>
+
+  <readouts>
+    <readout name="DREndcapTubesRO">
+      <segmentation type="CartesianGridXY" 
+             grid_size_x="1*mm" 
+             grid_size_y="1*mm"/>
+      <!-- NO SPACES BETWEEN ID DESCRIPTORS!!! -->
+      <id>stave:10,tower:6,air:1,col:16,row:16,clad:1,core:1,cherenkov:1</id>
```

Comment:
This should start with `system:6` (or `:5`?) and maybe also `side:-2`?

Instead of air/clad/core/cherenkov with one bit, why not 2 bits (of they are exclusive), or 4 bits to encode the flag?

Reply:
Right! I included a new `system:5` id and assigned it to the highest volume in the geometry node.
`side:-2` is not needed here because in the SDAction I check if the step is in endcap right (z>0) or left (z<0) and two hit collections (left and right) are used to store hits from the two endcaps, so there is no need for the `side` id in the hit-id.
About condensing 1-bit IDs into a larger id, it is certainly possible. However, in this implementation I have to use g4 copynumbers, retrieve them in the SDAction, and from them create the 64-bit id. If I condense more ids into a larger id, I have to include more functions in the geometry cpp code to encapsulate these short ids into a larger one and assign it to volumes as g4 copynumbers. The code will become longer and less readable both in the geometry part and in the SDAction part. If it is not a problem, I would prefer to keep this as simple as possible.

Reply:
@andresailer  can we resolve this?

Reply:
If you are convinced the two sides never end up in the same collection, yes.

Reply:
Yes, here is the code that separates left and right hits in the correct collections. Thanks! https://github.com/key4hep/k4geo/pull/411/files#diff-ec6c4af548e901207b2fbad9d312c13896804af83afaa86aa7a27872fb03b625R191-R201

Reply:
Ok, thanks. Resolving for now, we will see when using it in subsequent algorithms if it is annoying to have them in two separated collections. Indeed we usually have one collection for both endcaps.